src/server/helpers/github.js

```diff
@@ -11,31 +11,35 @@ import { conf } from '../helpers/config';
 const GITHUB_API_ENDPOINT = conf.get('GITHUB_API_ENDPOINT');
 const HTTP_PROXY = conf.get('HTTP_PROXY');
 
-export const requestGitHub = (options, callback) => {
-  const params = {
-    ...options,
-    proxy: HTTP_PROXY
-  };
-  params.uri = GITHUB_API_ENDPOINT + params.uri;
-  if (!params.headers) {
-    params.headers = {};
-  }
-  params.headers['User-Agent'] = 'SnapcraftBuild';
-  return request(params, callback);
+export const requestGitHub = options => {
+  return new Promise((resolve, reject) => {
+    const params = {
+      ...options,
+      proxy: HTTP_PROXY
+    };
+    params.uri = GITHUB_API_ENDPOINT + params.uri;
+    if (!params.headers) {
+      params.headers = {};
+    }
+    params.headers['User-Agent'] = 'SnapcraftBuild';
+    request(params, (err, response) => {
```

Comment:
I was wondering if using fetch from `isomorphic-fetch` wont be simpler and more consistent (as it is already based on promises), but as it's server side only and we have it set up to work with proxy it's probably not worth changing.